lib/src/common/managed_stream_subscription.dart

```diff
@@ -0,0 +1,42 @@
+import 'dart:async';
+
+/// An implementation of StreamSubscription that provides a didCancel future.
+class ManagedStreamSubscription<T> extends StreamSubscription<T> {
```

Comment:
If `StreamSubscription` were to change out from under us, though, we'd end up with weird bugs instead of static errors.